proto/regen/ecocredit/v1/tx.proto

```diff
@@ -305,3 +326,95 @@ message MsgUpdateClassMetadata {
 
 // MsgUpdateClassMetadataResponse is the MsgUpdateClassMetadata response type.
 message MsgUpdateClassMetadataResponse {}
+
+// MsgAddCreditType is the Msg/AddCreditType request type.
+// The credit types are immutable and cannot be deleted.
+message MsgAddCreditType {
+  // credit_types is a list of new credit types to add
+  repeated CreditType credit_types = 1;
+
+  // root_address is the address of the caller.
+  // this MUST equal the address of the gov module for the tx to succeed.
+  string root_address = 2;
+}
+
+// MsgAddCreditTypeResponse is the Msg/AddCreditType response type.
+message MsgAddCreditTypeResponse {}
+
+
+// MsgToggleAllowListRequest is the Msg/ToggleAllowListRequest request type
+message MsgToggleAllowListRequest {
+  // root_address is the address of the caller.
+  // this MUST equal the address of the gov module for the tx to succeed.
+  string root_address = 1;
+
+  // toggle is what the allow list will be set to.
+  bool toggle = 2;
+}
```

Comment:
```suggestion
// MsgToggleAllowListRequest is the Msg/ToggleAllowListRequest request type
message MsgToggleAllowListRequest {
  // root_address is the address of the signer.
  // This MUST equal the address of the gov module for the tx to succeed.
  string root_address = 1;

  // toggle is what the allow list will be set to.
  bool toggle = 2;
}
```

Reply:
Might also be nice to consistently put root address first or last in these message.